### PR TITLE
*: pin go1.20

### LINKED ISSUE #40969: upgrade go 1.20
## Enhancement

The latest Go release, version 1.20, arrives six months after [Go 1.19](https://tip.golang.org/doc/go1.19). Most of its changes are in the implementation of the toolchain, runtime, and libraries. As always, the release maintains the Go 1 [promise of compatibility](https://tip.golang.org/doc/go1compat).

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #40969

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: util/misc.go [hunk 1]
```diff
@@ -559,13 +559,12 @@ func IsTLSExpiredError(err error) bool {
...
 			return false
 		}
-		// TODO: remove it after upgrading 1.20
-		// case *tls.CertificateVerificationError:
-		// 	invali, ok := inval.Err.(x509.CertificateInvalidError)
-		// 	if !ok || invali.Reason != x509.Expired {
-		// 		return false
-		// 	}
-		// 	return true
+	case *tls.CertificateVerificationError:
```
Comment (hawkingrei) [inline util/misc.go:562]: It is related with go1.20. 

For a handshake failure due to a certificate verification failure, the TLS client and server now return an error of the new type [CertificateVerificationError](https://tip.golang.org/pkg/crypto/tls/#CertificateVerificationError), which includes the presented certificates.

**Review by lance6716 — APPROVED** [review]

**Comment by hawkingrei** [comment]
waiting for merging this [PR](https://github.com/PingCAP-QE/ci/pull/1815).

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**